compiler/rustc_resolve/src/late.rs

```diff
@@ -3934,12 +3934,14 @@ impl<'a, 'ast, 'ra: 'ast, 'tcx> LateResolutionVisitor<'a, 'ast, 'ra, 'tcx> {
                         // Move up the non-overlapping bindings to the or-pattern.
                         // Existing bindings just get "merged".
                         let collected = bindings.pop().unwrap().1;
+                        #[allow(rustc::potential_query_instability)] // FIXME
```

Comment:
How much effort is it to change `bindings` to `&mut SmallVec<[(PatBoundCtx, UnordSet<Ident>); 1]>`?

Reply:
I run into trouble here: https://github.com/rust-lang/rust/blob/085312b93e0384b27285e2b08612bba30428a184/compiler/rustc_resolve/src/late.rs#L3936-L3938
With the change, `collected` is an `UnordSet`, which does not implement `IntoIterator`, and so cannot be passed to `extend`. (Apologies if I am doing something stupid.)

Reply:
so we've got
```rust
pub trait ExtendUnord<T> {
    /// Extend this unord collection with the given `UnordItems`.
    /// This method is called `extend_unord` instead of just `extend` so it
    /// does not conflict with `Extend::extend`. Otherwise there would be many
    /// places where the two methods would have to be explicitly disambiguated
    /// via UFCS.
    fn extend_unord<I: Iterator<Item = T>>(&mut self, items: UnordItems<T, I>);
}
```
I think what we want is a method on `UnordSet` which gives u `UnordItems` without triggering the lint.

Reply:
It looks like `PatternBindings` now uses an `FxIndexMap` in master, so this may be a moot point: https://github.com/rust-lang/rust/blob/2c1ac85679678dfe5cce7ea8037735b0349ceaf3/compiler/rustc_resolve/src/late.rs#L124

Here is the relevant commit: https://github.com/rust-lang/rust/commit/30a0ac66dbb15ee8dd3951499b34df48e1d758a0